flow/src/main/java/flow/Dispatcher.java

```diff
@@ -23,6 +23,15 @@
    * Called when the history is about to change.  Note that Flow does not consider the
    * Traversal to be finished, and will not actually update the history, until the callback is
    * triggered. Traversals cannot be canceled.
+   * <p>
+   * Also called immediately after {@link Flow#setDispatcher}, to update the new dispatcher
+   * on Flow's current state. Such bootstrap Traversals have a null {@link Traversal#origin},
+   * and {@link Direction#REPLACE} as their direction.
+   * <p>
+   * Because of this bootstrapping behavior, Dispatchers are required to be idempotent. They
+   * should compare the {@link Flow#getKey(android.view.View) key of the current view(s)} to those
+   * in {@link Traversal#destination} before making visible updates. If no update is needed,
```

Comment:
I'd add some general advice about checking whether the app is already in the correct state for the incoming key (including the `getKey` check) before making any state changes.

It's possible that an app author might have other state to check/update.